miden-tx/src/testing/tx_context.rs

```diff
@@ -211,33 +216,369 @@ impl TransactionContextBuilder {
-                vec![consumed_note_1, consumed_note_2]
-            },
-            AssetPreservationStatus::PreservedWithAccountVaultDelta => {
-                vec![consumed_note_1, consumed_note_2, consumed_note_5]
-            },
-            AssetPreservationStatus::TooManyFungibleInput => {
-                vec![consumed_note_1, consumed_note_2, consumed_note_3]
-            },
-            AssetPreservationStatus::TooManyNonFungibleInput => {
-                vec![consumed_note_1, consumed_note_2, consumed_note_4]
-            },
-        };
-
-        self.input_notes(notes).expected_notes(output_notes)
+    /// Creates a new output [Note] for the transaction corresponding to this context.
+    fn add_output_note(
+        &mut self,
+        inputs: impl IntoIterator<Item = Felt>,
+        assets: impl IntoIterator<Item = Asset>,
+    ) -> Note {
+        let note = NoteBuilder::new(self.account.id(), &mut self.rng)
+            .note_inputs(inputs)
+            .expect("The inputs should be valid")
+            .add_assets(assets)
+            .build(&self.assembler)
+            .expect("The note details should be valid");
+
+        self.expected_output_notes.push(note.clone());
+        note
+    }
+
+    fn input_note_simple(
+        &mut self,
+        sender: AccountId,
+        assets: impl IntoIterator<Item = Asset>,
+        inputs: impl IntoIterator<Item = Felt>,
+    ) -> Note {
+        NoteBuilder::new(sender, ChaCha20Rng::from_seed(self.rng.gen()))
+            .note_inputs(inputs)
+            .unwrap()
+            .add_assets(assets)
+            .build(&self.assembler)
+            .unwrap()
+    }
+
+    fn input_note_with_one_output_note(
+        &mut self,
+        sender: AccountId,
+        assets: impl IntoIterator<Item = Asset>,
+        inputs: impl IntoIterator<Item = Felt>,
+        output: &Note,
+    ) -> Note {
+        let code = format!(
+            "
+            begin
+                # NOTE
+                # ---------------------------------------------------------------------------------
+                push.{recipient}
+                push.{PUBLIC_NOTE}
+                push.{aux}
+                push.{tag}
+                call.{ACCOUNT_CREATE_NOTE_MAST_ROOT}
+
+                push.{asset} movup.4
+                call.{ACCOUNT_ADD_ASSET_TO_NOTE_MAST_ROOT}
+                dropw dropw dropw
+            end
+            ",
+            PUBLIC_NOTE = NoteType::Public as u8,
+            recipient = prepare_word(&output.recipient().digest()),
+            aux = output.metadata().aux(),
+            tag = output.metadata().tag(),
+            asset = prepare_assets(output.assets())[0],
+        );
+
+        NoteBuilder::new(sender, ChaCha20Rng::from_seed(self.rng.gen()))
+            .note_inputs(inputs)
+            .unwrap()
+            .add_assets(assets)
+            .code(code)
+            .build(&self.assembler)
+            .unwrap()
+    }
```

Comment:
NOTE: I'm not sure what is the point of these notes. It looks like it should be a tx_script instead? There are quite a bit of weirdness to these:

- the `inputs` are never used
- the input `assets` are never consumed
- the output could have more than one asset, which is not handled here

These are largely also issues with `output_notes_data_procedure`.

Here too more work is needed to figure out how to handle these.